src/key.rs

```diff
@@ -364,14 +365,27 @@ impl PublicKey {
     /// the result would be the point at infinity, i.e. we are adding this point
     /// to its own negation
     pub fn combine(&self, other: &PublicKey) -> Result<PublicKey, Error> {
+        PublicKey::combine_keys(&[self, other])
+    }
+
+    /// Adds the keys in the provided slice together, returning the sum. Returns
+    /// an error if the result would be the point at infinity, i.e. we are adding
+    /// a point to its own negation
+    #[cfg(all(feature = "std"))]
+    pub fn combine_keys(keys: &[&PublicKey]) -> Result<PublicKey, Error> {
+        use std::mem::transmute;
+        use std::i32::MAX;
```

Comment:
Can you use `core` and drop the feature gate?

Reply:
Done thanks that's much better indeed.